Source/Charts/Data/Implementations/Standard/BarChartDataSet.swift

```diff
@@ -149,6 +156,26 @@ open class BarChartDataSet: BarLineScatterCandleBubbleChartDataSet, BarChartData
     /// the alpha value (transparency) that is used for drawing the highlight indicator bar. min = 0.0 (fully transparent), max = 1.0 (fully opaque)
     open var highlightAlpha = CGFloat(120.0 / 255.0)
     
+    /// array of gradient colors [[color1, color2], [color3, color4]]
+    open var barGradientColors: [[NSUIColor]]?
+    
+    open var barGradientOrientation: BarChartDataSet.BarGradientOrientation = .vertical
+    
+    /// - returns: The gradient colors at the given index of the DataSet's gradient color array.
+    /// This prevents out-of-bounds by performing a modulus on the gradient color index, so colours will repeat themselves.
+    open func barGradientColor(atIndex index: Int) -> [NSUIColor]?
```

Comment:
Swift naming convention would be `func barGradientColor(at index: Int) -> [NSUIColor]?`